addons/mass_mailing/models/mailing.py

```diff
@@ -232,6 +232,9 @@ def _compute_mailing_domain(self):
 
     @api.model
     def create(self, values):
+        # Get subject from "sms_subject" field when SMS installed
+        if values.get('mailing_type') == 'sms' and values.get('sms_subject'):
+            values['subject'] = values['sms_subject']
```

Comment:
This is not modular because the "sms" mailing_type doesn't exist in the mass_mailing module.
This code should be placed into mass_mailing_sms instead.

Reply:
My comment here above seems not addressed?
The code checking the sms_subject should be in the mass_mailing_sms module since the fields is defined there.

As a common rule, never use a field that is defined in a sub-module.
Only use a field when it's defined by your module or one of its dependencies.